### PR TITLE
feat(fast-near) View state dump to redis

### PR DESCRIPTION
**Pull request by mhalambek**
Port vgrichina code for dumping account info and contract data
into redis.
Also add test for checking if the redis dump works.

### CODE REVIEW

**Comment by mhalambek** [comment]
Major TODO, spin up redis instances during testing phase so redis tests can work.

File: tools/state-viewer/Cargo.toml [hunk 1]
```diff
@@ -18,6 +18,7 @@ serde = "1"
 serde_json = "1"
 tempfile = "3"
 tracing = "0.1.13"
+redis = "*"
```
Comment (nikurt) [inline tools/state-viewer/Cargo.toml:21]: Please pick a concrete version.

File: tools/state-viewer/src/state_dump.rs [hunk 2]
```diff
@@ -114,6 +117,59 @@ pub fn state_dump(
...
+
+    for (shard_id, state_root) in state_roots.iter().enumerate() {
+        let trie =
+            runtime.get_trie_for_shard(shard_id as u64, last_block_header.prev_hash()).unwrap();
+        let trie = TrieIterator::new(&trie, &state_root).unwrap();
+        for item in trie {
+            let (key, value) = item.unwrap();
+            if let Some(sr) = StateRecord::from_raw_key_value(key, value) {
+                if let StateRecord::Account { account_id, account } = &sr {
+                    println!("Account: {}", account_id);
```
Comment (nikurt) [inline tools/state-viewer/src/state_dump.rs:140]: Will this `println` millions of lines to stdout?

File: tools/state-viewer/src/state_dump.rs [hunk 3]
```diff
@@ -551,4 +607,52 @@ mod test {
...
+        let records_file = tempfile::NamedTempFile::new().unwrap();
+        use crate::state_dump::state_dump_redis;
+        let res = state_dump_redis(
+            runtime,
+            &state_roots,
+            last_block.header().clone(),
+            &near_config,
+        );
+        assert_eq!(res, Ok(()));
+        // let new_genesis = new_near_config.genesis;
```
Comment (nikurt) [inline tools/state-viewer/src/state_dump.rs:653]: Remove?

**Review by nikurt — APPROVED** [review]

File: tools/state-viewer/Cargo.toml (hunk 1 above)
Comment (mhalambek) [reply tools/state-viewer/Cargo.toml]: Picked it out

File: tools/state-viewer/src/state_dump.rs (hunk 3 above)
Comment (mhalambek) [reply tools/state-viewer/src/state_dump.rs]: Removed

**Review by rtsai123 — APPROVED** [review]

**Pull request merged**